### ISSUE TITLE
loss 관련 질의사항

### ISSUE THREAD

**Issue by aqaqsubin** [haven-jeon/KoGPT2-chatbot#18]
안녕하세요! loss를 계산하는 데 궁금한 점이 있어 issue를 작성하게 되었습니다.   

```
def training_step(self, batch, batch_idx):
    token_ids, mask, label = batch
    out = self(token_ids)
    mask_3d = mask.unsqueeze(dim=2).repeat_interleave(repeats=out.shape[2], dim=2)
    mask_out = torch.where(mask_3d == 1, out, self.neg * torch.ones_like(out))
    loss = self.loss_function(mask_out.transpose(2, 1), label)
    loss_avg = loss.sum() / mask.sum()
    self.log('train_loss', loss_avg)
    return loss_avg
```

위에 코드에서 answer가 아닌 tensor는 neg 값을 곱해 마스킹하는 것으로 이해했습니다. ​
cross entropy가 자연로그 함수를 사용하는데 negative 값을 곱해 마스킹하는 이유를 여쭤보고 싶습니다.

마스킹의 의도가 answer에 해당하는 tensor에 대한 loss만 계산하기 위함이라고 생각했는데,
위의 코드는 answer가 아닌 tensor의 loss도 포함하는 것 같습니다.

제가 잘못 이해한 부분이 있다면 알려주시길 바랍니다.

감사합니다.

**Comment by haven-jeon** [comment]
negative logits 값을 어사인 하기 위함이구요. 
아래 코드 부분과 동일하지는 않지만 의도는 같습니다. 

https://github.com/google-research/google-research/blob/e02792ead7866262ee2ace6088160ea5e4d06a5f/schema_guided_dst/baseline/train_and_predict.py#L592

나머지는 #17 참고해주세요.

**Comment by aqaqsubin** [comment]
네! 답변 주셔서 감사합니다:)
